packages/node_modules/@node-red/runtime/lib/flows/Flow.js

```diff
@@ -209,7 +211,9 @@ class Flow {
                                     node
                                 );
                                 this.subflowInstanceNodes[id] = subflow;
+                                subflowInstance[id] = subflow;
                                 subflow.start();
+                                delete subflowInstance[id];
```

Comment:
This will mean a call to `getNode` will behave differently if called during startup or called later on. That is not good.

`subflowInstance` should be tidied up during the `stop` phase.

Reply:
A link to the UI Module prototype. [ui_module code](https://github.com/node-red/node-red-dashboard/blob/2ef4cf00ebf7cdcd6c0722000e9cd4c468d5f9d2/ui.js#L137)
Cannot get subflow instance when subflow instance is placed in subflow.
For this reason, I commented it out in the prototype version.

Reply:
Okay - I was hoping to use the code to help understand what you want to do - but there is no code there.

```
     //console.log("# Belongs to subflow:", subflow.path);
```

What do you need to do with `subflow.path` ? I need to know exactly how you will be using that property. If this is explained in the design, please point me to exactly where.

Reply:
Please refer to pages 2-3 of the following document.
[08-Add_features_for_UI_Module.pptx](https://github.com/node-red-hitachi/node-red-notes/blob/master/Teleconference/20201124/08-Add_features_for_UI_Module.pptx)

Reply:
Thank you for the link. We would normally use the design notes to cover the details and *not* .pptx files.

From slide 3:
>  In the display of a UI module, we need information on which sub-flow instances (back to the topmost parent) the widgets belonging to the UI module belong to, so we need to be able to get values like the following.

I think this means you will be parsing the `path` property to get all of the subflow ids in the hierarchy? If so, that does mean the format of `path` property becomes fixed and we can't change it in the future.

That is not a problem - but it is something we need to acknowledge to make sure the format is suitable to never be changed again.

Reply:
Thank you for your confirmation.
Initially, I was considering getting the value by tracing the hierarchy, but I am using it because the value of the path property fits the purpose of the ui module.
There is no problem if there is a function that can obtain the same result.

Reply:
> Initially, I was considering getting the value by tracing the hierarchy, but I am using it because the value of the path property fits the purpose of the ui module.

I would feel better if we could see the actual code that will use the path property to help understand exactly what you are doing. Do you have that code written anywhere?

Reply:
An example of using the path property of subflow information has been added to the prototype version.
Since it is difficult to implement the display on the dashboard, the code example is a sample that outputs the order value in the group of each layer to the console.
[ui module: Added order acquisition code example for widgets under subgroup](https://github.com/node-red/node-red-dashboard/pull/621/commits/8f485688bf46b7d05a70947ac309493e8ccf5ad3)  

Output example:
```
# Widget: 3e487825.c62e78 ui_button
  Belongs to Subflow instalce : f5b5334f.5e91e/48cd027e.f6766c/a1ceb69a.073368
  subflow instance= a1ceb69a.073368
  widget or subgroup instance= 3e487825.c62e78    <---  'widget' id
  order= 1    <--- Get ’widget’ order value from subgroup template
  subflow instance= 48cd027e.f6766c
  widget or subgroup instance= dd83a85d.f75128    <---  'subgroup instance' id
  order= 0    <--- Get 'subgroup instance' order value  from subgroup template
  subgroup instance= a54f69c2.fd7c98
  order= 3    <--- Get the order value of'subgroup instance'from its own order property
```